docs/content/preview/architecture/transactions/read-restart-error.md

```diff
@@ -15,26 +15,68 @@ rightNav:
+
+How does YugabyteDB mitigate this clock skew anomaly?
+
+* First, note that the clock skew between all nodes in the cluster is always in a [max_clock_skew_usec](../../../reference/configuration/yb-tserver/#max-clock-skew-usec) bound due to clock synchronization algorithms.
+* For records with a commit timestamp later than `T1` + `max_clock_skew`, the database can be sure that these records were written after the read was issued. But for records with a commit timestamp between `T1` and `T1` + `max_clock_skew`, the tablet servers can find themselves in an ambiguous situation, such as the following:
+
+  * A read operation must observe all the commits that were acknowledged before the read was issued, even when the read time is picked on a node whose clock runs behind the node that committed the prior write. NOTE: the commit happened before the read as perceived by the user and not necessarily according to the machine clocks.
+
+  * A read operation must return a consistent snapshot of data, i.e. the read cannot simply advance its read time on observing a record with higher timestamp since the read already returned records from an older snapshot thus far. Therefore, the read must be restarted from the beginning with the advanced timestamp. Thus, the name read restart error.
+
+  * A read operation is free to include or exclude records committed by concurrent transactions. Implying, whenever the commit timestamp is lower than the read timestamp, the read is allowed to observe writes from the committed transaction, even when the commit happened concurrently with the read. Therefore, there is no ambiguity when the commit time is lower.
 
-  Note that the clock skew between all nodes in the cluster is always in a [max_clock_skew_usec](../../../reference/configuration/yb-tserver/#max-clock-skew-usec) bound due to clock synchronization algorithms.
-* For writes with a write timestamp later than `T1` + `max_clock_skew`, the database can be sure that these writes were done after the read timestamp had been chosen. But for writes with a write timestamp between `T1` and `T1` + `max_clock_skew`, node `N2` can find itself in an ambiguous situation, such as the following:
+* Whenever a read operation finds records with timestamp in the range `(T1, T1+max_clock_skew]`, to avoid breaking the strong guarantee that a reader should always be able to read what was committed earlier, and to read a consistent snapshot, the read operation raises a `Read restart` error to restart the read.
 
-  * It should return the data even if the client issued the read from a different node after writing the data, because the following guarantee needs to be maintained: the database should always return data that was committed in the past (past refers to the user-perceived past, and not based on machine clocks).
+{{<note title="Causal Reverse Anomaly">}}
```

Comment:
What is a causal reverse anomaly? First mention. This seems out of place, why are you mentioning it here?
Or is this entire scenario a 'causal reverse anomaly', in which case why not call it that earlier?

Reply:
It is another anomaly that stems from clock skew. We do not handle this anomaly. This is a piece of info for those who are curious. But I do agree that there is no point in mentioning this since readers who know about this anomaly already know that YB does not handle it.